### PR TITLE
add ci on the docker and fix main ci

### PR DESCRIPTION
**Pull request by teruto725**
Creating CI

Issue
#26 

Did 
- write ci don't use docker
  - .github/workflows/ci_on_scratch.yml
- write ci using docker
  - .github/workflows/ci_on_docker.yml

Caution
- Not work on local using act
  - maybe we have to some command install before run these yaml.

### CODE REVIEW

File: .github/workflows/ci_on_scratch.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+jobs:
+  run-test:
+    name: Run Test on Scratch
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Install yarn
+        working-directory: ./frontend
+        run: npm install -g yarn
```
Comment (wowry) [inline .github/workflows/ci_on_scratch.yml:15]:
```suggestion
```

Comment (wowry) [reply .github/workflows/ci_on_scratch.yml:15]: I wonder if this can be removed.

File: .github/workflows/ci_on_scratch.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Install yarn packages
+        working-directory: ./frontend
+        run: yarn install
```
Comment (wowry) [inline .github/workflows/ci_on_scratch.yml:18]:
```suggestion
        run: yarn install --frozen-lockfile
```

Comment (wowry) [reply .github/workflows/ci_on_scratch.yml:18]: In CI, can you add `--frozen-lockfile` option?
https://qiita.com/akameco/items/8945e434da315a0fbe44

File: .github/workflows/ci_on_scratch.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Install pip packages
+        working-directory: ./backend
+        run: python -m pip install -r requirements.txt
+      - name: pre commit run
+        uses: pre-commit/action@v2.0.3
+      - name: test frontend
+        working-directory: ./frontend
+        run: yarn test --watchAll=false
```
Comment (wowry) [inline .github/workflows/ci_on_scratch.yml:26]:
```suggestion
        run: CI=true yarn test
```

Comment (wowry) [reply .github/workflows/ci_on_scratch.yml:26]: In CI, I'm actually assuming that CI option will be added.
https://github.com/ani-hovhannisyan/kanji-visualization/pull/23

However, `yarn test --watchAll=false` may be also OK.

File: .github/workflows/ci_on_docker.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      # -d でバックグラウンド実行
+      - name: docker build
+        run: docker-compose up -d
+      # -Tは no TTY への対策
+      - name: frontend test
+        run: docker-compose exec -T frontend yarn test --watchAll=false
```
Comment (wowry) [inline .github/workflows/ci_on_docker.yml:18]:
```suggestion
        run: docker-compose exec -T -e CI=true frontend yarn test
```

Comment (wowry) [reply .github/workflows/ci_on_docker.yml:18]: Same as ci_on_scratch.

File: .github/workflows/ci_on_scratch.yml (hunk 1 above)
Comment (teruto725) [reply .github/workflows/ci_on_scratch.yml]: Yes you can. Actuary I added for the local environment as act docker. But finally it didn't work, so Let's delete

**Comment by wowry** [comment]
LGTM! Thanks!! Just a few minor remarks above.

File: .github/workflows/ci_on_scratch.yml (hunk 3 above)
Comment (teruto725) [reply .github/workflows/ci_on_scratch.yml:23]: Sorry for missing your tips.
I'll update.

**Comment by teruto725** [comment]
I fixed following your comments
check please!
@wowry

**Review by wowry — APPROVED** [review]
Thank you very much for the quick response!

**Pull request merged**